components/script/dom/htmlinputelement.rs

```diff
@@ -408,8 +408,19 @@ impl HTMLInputElementMethods for HTMLInputElement {
     fn SetValue(&self, value: DOMString) -> ErrorResult {
         match self.value_mode() {
             ValueMode::Value => {
+                // Step 1.
+                let old_value = self.textinput.borrow().single_line_content().clone();
```

Comment:
Why clone?

Why do you clone..?

```rust
// Steps 1-2.
let old_value = self.textinput.borrow_mut().replace(value);
```

Reply:
Because in step 5 we need to compare the new value against the old value.